src/Data/Connection/AbstractConnectionResolver.php

```diff
@@ -306,6 +313,17 @@ public function set_query_arg( $key, $value ) {
 		return $this;
 	}
 
+	/**
+	 * Sets the connection Query class.
+	 *
+	 * @param string $class  Query class name.
+	 *
+	 * @return void
+	 */
+	public function set_query_class( $class ) {
```

Comment:
I've got concerns/questions from a "holistic" POV - _nothing_ that follows should be considering blocking since the same issues exist with the current method of setting the query class via `WPConnectionType`:

I love the fact that this pattern makes local overloading in a resolver possible, and brings Comment/User connection queries the same extensibility patten, but I'm concerned about it's availability on `AbstractConnectionResolver`, when only 4 (including those in this PR) out of the bundled 13 Connection resolvers actually use a `$query_class`. 

Similarly, I'm concerned around the lack of checks around the provided `$query_class`. We dont want to enforce a `WP_Query` child, but we do need to make sure that the provided class string (`SWP_Query`, `WC_Query`, `Tribe_Query`, `FWP_Query`, etc etc) is actually compatible with the Connection resolver relying on it.

The latter issue can be solved with an overloadable `is_valid_query_class( string $class )`  that could for example check for `method_exists( $class, 'query' )`. 

We could even in theory use the same `is_valid_query_class()` to throw an exception if `! empy( $this->query_class )` on the Connection Resolvers that don't use it (would need to figure out backwards-compatibility), but I think we should first confirm that `AbstractConnectionResolver` is really the ideal home for this instead of adding what's _possibly_ tech debt to the majority of the included connection resolvers.

Reply:
Actually it's more like 6 or 7 out of 13, because they extend one of the 4 connection resolvers and not the AbstractConnectionResolver class directly, Some of them don't even have a `get_query()` function and default to parent `get_query()` function in one of the 4. I agree there should be `class_exists( $this->query_class )` check somewhere and a `is_valid_query_class()` function is a good idea. I'll add it immediately.

Reply:
> because they extend one of the 4 connection resolvers and not the AbstractConnectionResolver class directly, Some of them don't even have a get_query() function and default to parent get_query() function in one of the 4.

Actual numbers aside ☝️is the crux of my concern: we're exposing a setter that semantically should apply on any connection but will be ignored by most of them - something that becomes less obvious from the code the better we are at DRYing out the implementing classes. But I don't have a good alternative 🤔

Reply:
Or breaking up AbstractConnectionResolver into traits, but I think either is bad DX in this particular situation since it extenders then need to know which intermediate class / trait is right for a particular WP type, but I don't see a way to have those classes/traits to use existing WP namings. 🤷‍♂️

Reply:
Exactly, but the connection resolvers not using `$query_class` are less likely to be extended for usage. `EnqueuedScripts`, `EnqueuedStylesheet`, `Plugin`, `Taxonomy`, `Theme`, `UserRole`, and `ContentType`